src/System.Runtime.Extensions/src/System/IO/Path.cs

```diff
@@ -157,15 +157,19 @@ public static string GetFileNameWithoutExtension(string path)
 
         // Returns a cryptographically strong random 8.3 string that can be 
         // used as either a folder name or a file name.
-        public static string GetRandomFileName()
+        public static unsafe string GetRandomFileName()
         {
             // 5 bytes == 40 bits == 40/5 == 8 chars in our encoding.
             // So 10 bytes provides 16 chars, of which we need 11
             // for the 8.3 name.
-            byte[] key = CreateCryptoRandomByteArray(10);
+            const int KeyLength = 10;
+            byte* key = stackalloc byte[KeyLength];
+            GetCryptoRandomBytes(key, KeyLength);
 
             // rndCharArray is expected to be 16 chars
-            char[] rndCharArray = ToBase32StringSuitableForDirName(key).ToCharArray();
+            const int RndCharArrayLength = 16;
+            char* rndCharArray = stackalloc char[RndCharArrayLength];
```

Comment:
@jamesqo You should never mess with a string that you didn't create or have passed on only because it is expected to be immutable. We know what the scope of the pin is here- and it is exactly what happens in the string class as per my comment above (see `CtorCharPtrStartLength`).

In general you don't want to go there. With native interop and using pointers here already I don't see a problem with it, particularly as we're in the Path class and it's methods often get hit in tight loops. That said, this particular method shouldn't be called like that, and some others on the team might feel differently. :)